utility_functions.cpp

```diff
@@ -0,0 +1,42 @@
+#include <ibamr/app_namespaces.h>
+
+#include <ibtk/ibtk_utilities.h>
+
+#include <Box.h>
+
+// Local includes
+#include "utility_functions.h"
+
+namespace IBAMR
+{
+double
+convolution(double alpha,
+            const CellData<NDIM, double>& a_data,
+            double beta,
+            const CellData<NDIM, double>& b_data,
+            std::function<double(double)> phi,
+            unsigned int phi_width,
+            const CellIndex<NDIM>& idx,
+            const double* const dx)
+{
+    // Box over which integration is performed
+    Box<NDIM> int_box(idx, idx);
+    int_box.grow(phi_width);
+#ifndef NDEBUG
+    // Error checking. Need ghost data for patch data.
+    TBOX_ASSERT(a_data.getGhostBox() * int_box == int_box);
```

Comment:
Could you elaborate on why we need ghost data here, and what exactly this line is doing?

Reply:
This line computes the intersection between the two boxes, and makes sure that it's equal to the integration box. * is an overloaded operator for type `Box<NDIM>`. Essentially, we're checking that the integration box is contained within the ghost box.